HW3/P5/label_regions.cl

```diff
@@ -87,11 +111,20 @@ propagate_labels(__global __read_write int *labels,
         // We set new_label to the value of old_label, but you will need
         // to adjust this for correctness.
         new_label = old_label;
```

Comment:
After parts 2 and 4, you should use buffer[buf_w * buf_y + buf_x] instead of old_label.